### PR TITLE
Support coverage upload

### PR DESCRIPTION
**Pull request by omus**

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -42,3 +42,7 @@ build_script:
 test_script:
   - echo "%JL_TEST_SCRIPT%"
   - C:\julia\bin\julia -e "%JL_TEST_SCRIPT%"
+
+after_script:
```
Comment (simonbyrne) [inline appveyor.yml]: `on_success`? See https://www.appveyor.com/docs/build-configuration/

Comment (simonbyrne) [reply appveyor.yml]: Also, please make the same change to `README.md`

Comment (omus) [reply appveyor.yml]: Will do

**Comment by omus** [comment]
Testing these changes on Mocking.jl: https://ci.appveyor.com/project/omus/mocking-jl/build/1.0.342

**Comment by simonbyrne** [comment]
Feel free to add more tests to this repo as well.

**Comment by omus** [comment]
Can you add the Codecov GitHub integration? That should be able to verify we're testing the Appveyor coverage upload with the code I added.

**Comment by simonbyrne** [comment]
You should have admin access now

**Comment by omus** [comment]
PR should be good now. I added a Codecov badge to the README. Feel free to squash

**Comment by omus** [comment]
Noticed the code comment wasn't quite right

**Comment by simonbyrne** [comment]
My only concern is that it adds about 2 mins to each CI run. Is there a way to check from within Julia if codecov is enabled?

Alternatively, we could name the variable `JL_CODECOV_SCRIPT` (and maybe leave it commented out by default).

**Comment by omus** [comment]
I agree that it is problematic to add extra CI time for repos which may not need it. I think commenting it out in the README by default with a comment is the right direction. I'm not aware of a way to detect Codecov being enabled from within Julia. If it is possible I imagine it would be from within the Coverage.jl repo which takes some time to install.

**Comment by omus** [comment]
> Alternatively, we could name the variable `JL_CODECOV_SCRIPT` (and maybe leave it commented out by default).

I think I'll also do this as we may want to add a `JL_COVERALL_SCRIPT` variable as well to the `on_success` step.

**Comment by omus** [comment]
@simonbyrne anything left to do here?

**Comment by simonbyrne** [comment]
Thanks!

**Pull request merged**